bootstrap.py

```diff
@@ -1053,12 +1082,71 @@ def exec_service(service, command, failonerror=True):
         migrate_systems(options.org, options.activationkey)
         if options.enablerepos:
             enable_repos()
+    elif options.new_capsule:
+        # > ELIF new_capsule and foreman_fqdn set, will migrate to other capsule
+        #
+        # > will replace CA certificate, reinstall katello-agent, gofer
+        # > will optionally update hostgroup and location
+        # > wil update system definition to point to new capsule for content,
+        # > Puppet, OpenSCAP and update Puppet configuration (if applicable)
+        # > MANUAL SIGNING OF CSR OR MANUALLY CREATING AUTO-SIGN RULE STILL REQUIRED!
+        # > API doesn't have a public endpoint for creating auto-sign entries yet!
+
+        # Make system ready for switch, gather required data
+        get_bootstrap_rpm(clean=True, unreg=False)
+        install_katello_agent()
+        API_PORT = get_api_port()
+        capsule_id = return_matching_foreman_key('smart_proxies', 'name="%s"' % options.foreman_fqdn, 'id', False)
+        host_id = return_matching_foreman_key('hosts', 'name="%s"' % FQDN, 'id', False)
+        capsule_features = get_capsule_features(capsule_id)
+
+        # Optionally configure new hostgroup, location
+        if options.hostgroup:
+            print_running("Calling Foreman API to switch hostgroup for %s to %s" % (FQDN, options.hostgroup))
+            update_host_config('hostgroup', options.hostgroup, host_id)
+        if options.location:
+            print_running("Calling Foreman API to switch location for %s to %s" % (FQDN, options.location))
+            update_host_config('location', options.location, host_id)
+
+        # Configure new proxy_id for Puppet (if not skipped), and OpenSCAP (if available and not skipped)
+        if 'foreman' not in options.skip and 'puppet' not in options.skip:
+            print_running("Calling Foreman API to update Puppet master and Puppet CA for %s to %s" % (FQDN, options.foreman_fqdn))
+            update_host_capsule_mapping("puppet_proxy_id", capsule_id, host_id)
+            update_host_capsule_mapping("puppet_ca_proxy_id", capsule_id, host_id)
+        if 'foreman' not in options.skip and 'Openscap' in capsule_features:
+            print_running("Calling Foreman API to update OpenSCAP proxy for %s to %s" % (FQDN, options.foreman_fqdn))
+            update_host_capsule_mapping("openscap_proxy_id", capsule_id, host_id)
+        elif 'foreman' not in options.skip and 'Openscap' not in capsule_features:
+            print_warning("New capsule doesn't have OpenSCAP capability, not switching / configuring openscap_proxy_id")
+
+        print_running("Calling Foreman API to update content source for %s to %s" % (FQDN, options.foreman_fqdn))
+        update_host_capsule_mapping("content_source_id", capsule_id, host_id)
```

Comment:
At least in Katello 3.5/3.6, this needs to be `host[content_facet_attributes][content_source_id]`, not `host[content_source_id]`

Reply:
Ok, @evgeni, tiny problem.

I have changed the code so that it - in case we try to change content_source_id - it creates a special jdata that has both the old `content_source_id` in there as well as the new [content_facet_attributes][content_source_id].

From a hammer point of view, this allows you to see the changed content source. From the point of view of the Satellite UI though, it doesn't. It still shows you the first content source it used, or maybe it's the server that was used to register, I don't know.

All I know is that 

    $ hammer host info --id HOSTID

will show the new content source, and the UI will not.

This is only in Satellite 6.3: in Satellite 6.2, everything works just fine.

Reply:
Mhhh. I'll have a try on monday with my Katello 3.5. I *thought* (and that's why I told you that) that this hack worked for me.

Reply:
yeah, can reproduce. weird.

Reply:
memo to self, try to see what hammer does

Reply:
So `hammer -d host update --name centos7.kangae.example.com --content-source-id 2` shows

```
[ INFO 2018-01-22 10:55:53 API] Server: https://pipeline-katello-3-5-centos7.kangae.example.com
[ INFO 2018-01-22 10:55:53 API] PUT /api/hosts/20
[DEBUG 2018-01-22 10:55:53 API] Params: {
    "host" => {
                                 "name" => "centos7.kangae.example.com",
                   "compute_attributes" => {
            "volumes_attributes" => {}
        },
             "content_facet_attributes" => {
            "content_source_id" => "2"
        },
        "subscription_facet_attributes" => {},
           "host_parameters_attributes" => [],
                "interfaces_attributes" => []
    }
}
```

and the reply looks correct too:

```
         "content_facet_attributes" => {
                                    "id" => 18,
                                  "uuid" => "2acbdedf-46cc-4504-b0f3-e6a743f064f4",
                       "content_view_id" => 3,
                     "content_view_name" => "default-p4",
              "lifecycle_environment_id" => 1,
            "lifecycle_environment_name" => "Library",
                     "content_source_id" => 2,
                   "content_source_name" => "pipeline-proxy-3-5-centos7.kangae.example.com",
               "kickstart_repository_id" => nil,
                         "errata_counts" => {
               "security" => 0,
                 "bugfix" => 0,
            "enhancement" => 0,
                  "total" => 0
        },
              "applicable_package_count" => 0,
              "upgradable_package_count" => 0,
                          "content_view" => {
              "id" => 3,
            "name" => "default-p4"
        },
                 "lifecycle_environment" => {
              "id" => 1,
            "name" => "Library"
        },
                        "content_source" => {
              "id" => 2,
            "name" => "pipeline-proxy-3-5-centos7.kangae.example.com",
             "url" => "https://pipeline-proxy-3-5-centos7.kangae.example.com:9090"
        },
                  "content_view_version" => "1.0",
               "content_view_version_id" => 3,
                 "content_view_default?" => false,
        "lifecycle_environment_library?" => true,
               "katello_agent_installed" => true
    },
```

yet the UI still shows the wrong name